MediaBrowser.Providers/MediaInfo/MediaInfoResolver.cs

```diff
@@ -0,0 +1,218 @@
+        public async IAsyncEnumerable<MediaStream> GetExternalStreams(
+            Video video,
+            int startIndex,
+            IDirectoryService directoryService,
+            bool clearCache,
+            [EnumeratorCancellation] CancellationToken cancellationToken)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+
+            if (!video.IsFileProtocol)
+            {
+                yield break;
+            }
+
+            var pathInfos = GetExternalFiles(video, directoryService, clearCache);
+
+            foreach (var pathInfo in pathInfos )
+            {
+                string fileNameWithoutExtension = Path.GetFileNameWithoutExtension(pathInfo.Path);
+                Model.MediaInfo.MediaInfo mediaInfo = await GetMediaInfo(pathInfo.Path, _type, cancellationToken).ConfigureAwait(false);
+
+                if (mediaInfo.MediaStreams.Count == 1)
+                {
+                    MediaStream mediaStream = mediaInfo.MediaStreams.First();
+                    mediaStream.Index = startIndex++;
+                    mediaStream.IsExternal = true;
+                    mediaStream.Path = pathInfo.Path;
+                    mediaStream.IsDefault = pathInfo.IsDefault || mediaStream.IsDefault;
+                    mediaStream.IsForced = pathInfo.IsForced || mediaStream.IsForced;
+
+                    yield return MergeMetadata(mediaStream, pathInfo);
+                }
+                else
+                {
+                    foreach (MediaStream mediaStream in mediaInfo.MediaStreams)
+                    {
+                        mediaStream.Index = startIndex++;
+                        mediaStream.IsExternal = true;
+                        mediaStream.Path = pathInfo.Path;
+
+                        yield return MergeMetadata(mediaStream, pathInfo);
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Returns the external file infos for the given video.
+        /// </summary>
+        /// <param name="video">The video to search from.</param>
+        /// <param name="directoryService">The directory service to search for files.</param>
+        /// <param name="clearCache">True if the directory service cache should be cleared before searching.</param>
+        /// <returns>The external file paths located.</returns>
+        public IEnumerable<ExternalPathParserResult> GetExternalFiles(
+            Video video,
+            IDirectoryService directoryService,
+            bool clearCache)
+        {
+            if (!video.IsFileProtocol)
+            {
+                yield break;
+            }
+
+            // Check if video folder exists
+            string folder = video.ContainingFolderPath;
+            if (!Directory.Exists(folder))
+            {
+                yield break;
+            }
+
+            var files = directoryService.GetFilePaths(folder, clearCache, true);
```

Comment:
This would let you pick up files added by plugins to the internal metadata
```suggestion
            var files = directoryService.GetFilePaths(folder, clearCache).ToList();
            files.AddRange(directoryService.GetFilePaths(video.GetInternalMetadataPath(), clearCache));
```